### ISSUE TITLE
Read first draft

### ISSUE THREAD

**Issue by joachimvh** [comunica/Article-ISWC2019-AMF#1]
We have a first draft available, would it be possible to already go through it?

**Comment by rubensworks** [comment]
And also please give your suggestions on what can/should be removed, as we will have space issues :-) @mielvds @RubenVerborgh

**Comment by mielvds** [comment]
will do ✂️

**Comment by RubenVerborgh** [comment]
Very good research article, from beginning to end.

Attention points:
- We might want to emphasize the necessity more in the introduction. It might seem far-fetched otherwise. So explain clearly: SPARQL endpoints are expensive, TPF is cheap but slow. Is there something minimal we can do to improve speed, without becoming too expensive? That minimal thing is AMF. We test how you should configure it; answers in the conclusion, now flip that switch and querying is faster by x% at the expense of Y.
- The "discussion" section in the results is very good, but I would move to conclusions (as they would in other sciences where the conclusion is not just a summary).
- The results subsection is accurate, but dry and hard to follow because of that. You sometimes loose track of what is happening. x increases, y decreases, hypotheses are being rejected and accepted all over the place… but what does it concretely mean? good or bad? Try rephrasing these with just a hint of the discussions in there, without giving away too much. It's very hard to get through now.

I would like to read the results and conclusion again after edits.

Suggestions for cutting: besides some research questions (only if needed), I think we can/should have less on the software/implementation. I know, I know. But the important thing that should remain is: implementing this was easy, no modifications needed but just extra components and config. Actually, that isn't emphasized enough, so cut the rest to make this stand out.

Great work, guys!

**Comment by RubenVerborgh** [comment]
One nit though: several typos were highlighted by my spell checker. How did these get committed? Always type with your checker on; or even make it a commit hook.

**Comment by rubensworks** [comment]
Thanks for the review!

> I think we can/should have less on the software/implementation.

I'll see what's possible. In any case, the largest chunk is the evaluation section, so we'll have to do some cutting in there (and the corresponding research questions) in any case I'm afraid.

> several typos were highlighted by my spell checker. How did these get committed? 

It looks like my spell checker was disabled 😱

**Comment by rubensworks** [comment]
@RubenVerborgh What did you think about RQ 3, 5 and 6 (and their corresponding results)?
I would remove those, as they have the least interesting results IMO.

**Comment by RubenVerborgh** [comment]
The description of the results makes it a bit difficult to gauge interestingness (cfr. remarks above). My feeling is that they are sufficiently interesting, perhaps with the exception of 5. But all of them lead to good discussion points for me.

**Comment by rubensworks** [comment]
@RubenVerborgh Sure, most of them are interesting indeed :-)
But we have to cut unfortunately (unless you have another suggestion?),
so I think those 3 would be the least significant loss.
(We can still use them in a follow-up paper though)

**Comment by RubenVerborgh** [comment]
Let's first assess how far we get by truncating the implementation descriptions. The results will matter more to the ISWC audience than these (unfortunately perhaps, but that's the way it is, so we cannot influence that). Let's then look on (virtual) paper to assess how much needs to go. Because removing results also shrinks the contribution.

**Comment by rubensworks** [comment]
> Let's first assess how far we get by truncating the implementation descriptions.

Sounds good! Will ping you once that's done.
